crates/bevy_pbr/src/environment_map/mod.rs

```diff
@@ -29,90 +59,460 @@ impl Plugin for EnvironmentMapPlugin {
+    ) -> GpuImage {
+        let texture = render_device.create_texture(&TextureDescriptor {
+            label: match kind {
+                EnvironmentMapKind::Diffuse => Some("environment_map_diffuse_texture"),
+                EnvironmentMapKind::Specular => Some("environment_map_specular_texture"),
+            },
+            size: Extent3d {
+                width: extents.width,
+                height: extents.height,
+                depth_or_array_layers: self.cubemaps.len() as u32 * 6,
+            },
+            mip_level_count,
+            sample_count: 1,
+            dimension: TextureDimension::D2,
+            format: texture_format,
+            usage: TextureUsages::TEXTURE_BINDING | TextureUsages::COPY_DST,
+            view_formats: &[],
+        });
+
+        let texture_view = texture.create_view(&TextureViewDescriptor {
+            label: match kind {
+                EnvironmentMapKind::Diffuse => Some("environment_map_diffuse_texture_view"),
+                EnvironmentMapKind::Specular => Some("environment_map_specular_texture_view"),
+            },
+            format: Some(texture_format),
+            dimension: Some(ENVIRONMENT_MAPS_CUBE_DIMENSION),
+            aspect: TextureAspect::All,
+            base_mip_level: 0,
+            mip_level_count: Some(mip_level_count),
+            base_array_layer: 0,
+            array_layer_count: Some(self.cubemaps.len() as u32 * 6),
+        });
+
+        let sampler = render_device.create_sampler(&SamplerDescriptor {
+            label: match kind {
+                EnvironmentMapKind::Diffuse => Some("environment_map_diffuse_sampler"),
+                EnvironmentMapKind::Specular => Some("environment_map_specular_sampler"),
+            },
+            mag_filter: FilterMode::Linear,
+            min_filter: FilterMode::Linear,
+            mipmap_filter: FilterMode::Linear,
+            ..SamplerDescriptor::default()
+        });
+
+        GpuImage {
+            texture,
+            texture_view,
+            texture_format,
+            sampler,
+            size: vec2(extents.width as f32, extents.height as f32),
+            mip_level_count,
+        }
+    }
+
+    // Combines all cubemaps into a single cubemap array.
+    fn copy_cubemaps_in(
+        &self,
+        render_device: &RenderDevice,
+        render_queue: &RenderQueue,
+        kind: EnvironmentMapKind,
+    ) {
+        let environment_map_images = self
+            .images
+            .as_ref()
+            .expect("`copy_cubemaps_in()` called with no texture present");
+
+        let dest_image = match kind {
+            EnvironmentMapKind::Diffuse => &environment_map_images.diffuse,
+            EnvironmentMapKind::Specular => &environment_map_images.specular,
+        };
+
+        let (width, height) = (dest_image.size.x as u32, dest_image.size.y as u32);
+
+        let mut command_encoder = render_device.create_command_encoder(&CommandEncoderDescriptor {
```

Comment:
I like this idea and I can work on adding it. I think it makes sense to land it initially since it's an infrastructural change and adding `COPY_SRC` is ugly.

Reply:
Agree with rob. Will skip reviewing this part of the PR for now.

Reply:
As discussed on Discord this is now using binding arrays, so the extra GPU copies, as well as `COPY_SRC`, are gone.